Content.Shared/_DV/Item/ItemToggle/Components/ComponentTogglerComponent.cs

```diff
@@ -0,0 +1,18 @@
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Item.ItemToggle.Components;
+
+public sealed partial class ComponentTogglerComponent
+{
+    /// <summary>
+    /// Components that are removed on activation. Allows "swapping" components on toggle.
+    /// </summary>
+    [DataField]
+    public ComponentRegistry? ComponentsRemovedOnActivate;
+
+    /// <summary>
+    /// Components that are added on deactivation. Allows "swapping" components on toggle.
+    /// </summary>
+    [DataField]
+    public ComponentRegistry? ComponentsAddedOnDeactivate;
+}
```

Comment:
dont need 2 of the same thing, rename to RemovedComponents and use it for both cases
then remove the field-less one from the yml

Reply:
So true. Since upstream has 'RemoveComponents' (only used on deactivate, so not helpful here) I went with 'DeactivatedComponents'.